middleware.go

```diff
@@ -14,155 +11,63 @@ import (
 	"github.com/square/go-jose/v3/jwt"
 )
 
-//ContextKey used to access values in the context
-type ContextKey string
-
-const (
-	//ContextJWTError is used to access the JWT error in the context
-	ContextJWTError ContextKey = "JWT_ERROR"
-	//ContextJWTObject is used to access the JWT object in the context
-	ContextJWTObject ContextKey = "JWT_OBJECT"
-	//ContextJWTClaims is used to access the JWT claims in the context
-	ContextJWTClaims ContextKey = "JWT_CLAIMS"
-	//ContextValidatorError is used to access the error provided by validators
-	ContextValidatorError ContextKey = "VALIDATOR_ERROR"
-
-	//ErrInvalidToken is returned as a error if the token was invalid
-	ErrInvalidToken = "The token provided was invalid"
-)
-
-//JWTMiddleware is used to mange JWT middleware
-type JWTMiddleware struct {
-	JWKS        *jose.JSONWebKeySet
-	Issuer      string
-	EnableDebug bool
-}
+type JWTResolver func(w http.ResponseWriter, r *http.Request) (JWTInfo, error)
 
-//NewJWTMiddleware is used to greate a new JWTMiddleware
-func NewJWTMiddleware(jwks *jose.JSONWebKeySet, issuer string) JWTMiddleware {
-	return JWTMiddleware{
-		JWKS:        jwks,
-		Issuer:      issuer,
-		EnableDebug: false,
-	}
+type JWTInfo struct {
+	JWKS   *jose.JSONWebKeySet
+	Issuer string
 }
 
-//NewJWTMiddlewareFromOpenID creates a new JWTMiddleware from a OpenID definition
-func NewJWTMiddlewareFromOpenID(baseURL string) (*JWTMiddleware, error) {
-
-	resp, err := http.Get(baseURL + "/.well-known/openid-configuration")
-	if err != nil {
-		return nil, err
-	}
-
-	if resp.StatusCode != 200 {
-		return nil, errors.New(resp.Status)
-	}
-	byteArr, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, err
-	}
-
-	var bodyWellKnown OpenIDWellKnown
-	err = json.Unmarshal(byteArr, &bodyWellKnown)
-	if err != nil {
-		return nil, err
-	}
-
-	keyset, err := GetJWKSetFromOpenIDURL(bodyWellKnown.JWKSURL)
-	if err != nil {
-		return nil, err
-	}
-
-	middleware := NewJWTMiddleware(keyset, bodyWellKnown.Issuer)
-	return &middleware, nil
-
-}
-
-//GetJWKSetFromOpenIDURL creates a jsonwebkeyset from the JWK url provided
-func GetJWKSetFromOpenIDURL(JWKURL string) (*jose.JSONWebKeySet, error) {
-	resp, err := http.Get(JWKURL)
-	if err != nil {
-		return nil, err
-	}
-
-	if resp.StatusCode < 199 && resp.StatusCode < 301 {
-		return nil, errors.New("Response code was not sucessfull")
-	}
-
-	byteArr, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, err
-	}
-
-	var keySet jose.JSONWebKeySet
-
-	err = json.Unmarshal(byteArr, &keySet)
-
-	return &keySet, err
+type JWTMiddleware struct {
```

Comment:
Codacy found an issue: [exported type JWTMiddleware should have comment or be unexported](https://app.codacy.com/manual/johnfg10/gojwt/pullRequest?prid=5899569)